apps/emqx_audit/src/emqx_audit.erl

```diff
@@ -93,7 +99,7 @@ to_audit(#{from := From} = Log) when is_atom(From) ->
         %% request detail
         http_status_code = StatusCode,
         http_method = Method,
-        http_request = Request,
+        http_request = truncate_http_request(Request),
```

Comment:
I think this is too narrow, I'd be much more wary with this: if someone changes how this looks in the future, we'll likely forget to update this code.

My general question is: if this is basically for dashboard to consume as text, why don't we just `unicode:characters_to_binary(io_lib:format("~P", [Request, _Depth = 12], #{chars_limit => 10240}))` it before saving? Feels unnecessary to reimplement what the OTP already has in very well tested form. Especially given that we need some sort of [truncation for shell arguments](https://github.com/emqx/emqx/blob/46877e979b638ae17ec79464e7af0100ef4e0b72/apps/emqx_machine/src/emqx_restricted_shell.erl#L134-L139) anyway, which have arbitrary shapes.